pkg/controllers/binding/common.go

```diff
@@ -37,112 +36,98 @@ import (
 	"github.com/karmada-io/karmada/pkg/util/overridemanager"
 )
 
-// ensureWork ensure Work to be created or updated.
-func ensureWork(
-	ctx context.Context, c client.Client, resourceInterpreter resourceinterpreter.ResourceInterpreter, workload *unstructured.Unstructured,
-	overrideManager overridemanager.OverrideManager, binding metav1.Object, scope apiextensionsv1.ResourceScope,
-) error {
-	var targetClusters []workv1alpha2.TargetCluster
-	var placement *policyv1alpha1.Placement
-	var requiredByBindingSnapshot []workv1alpha2.BindingSnapshot
-	var replicas int32
-	var conflictResolutionInBinding policyv1alpha1.ConflictResolution
-	var suspension *policyv1alpha1.Suspension
-	switch scope {
-	case apiextensionsv1.NamespaceScoped:
+// workSyncer is design for syncing work objects as expected.
+type workSyncer struct {
+	client      client.Client
+	interpreter resourceinterpreter.ResourceInterpreter
+	overrider   overridemanager.OverrideManager
+}
```

Comment:
The `workSyncer` holds some immutable parameters, mainly used to reduce the number of parameters passed to ensureWork(). Is this the idea?

Reply:
It's like this.